app/components/channel_icon.js

```diff
@@ -90,30 +90,64 @@ export default class ChannelIcon extends React.PureComponent {
                 </View>
             );
         } else if (type === General.DM_CHANNEL) {
-            if (status === General.ONLINE) {
+            switch (status) {
+            case General.AWAY:
                 icon = (
-                    <OnlineStatus
+                    <AwayAvatar
                         width={size}
                         height={size}
-                        color={theme.onlineIndicator}
+                        color={theme.awayIndicator}
                     />
                 );
-            } else if (status === General.AWAY) {
+                break;
+            case General.DND: {
+                let dndIndicator;
+                if (theme.dndIndicator) {
```

Comment:
This section won't work because of the changes I made in https://github.com/mattermost/mattermost-redux/pull/311. I think we need to make an update to that PR to try to use the colours from whichever built-in theme you have specified instead of just using the default one.

Reply:
OK I definitely don't follow